### DISCUSSION TITLE
Proxy or port forwarding in Docker Swarm

### DISCUSSION THREAD

**Discussion by rm-rf-etc** [BretFisher/ama#156] [Q&A]
In k8s, kubectl provides `kubectl proxy` for accessing the k8s dashboard, and `kubectl port-forward` for accessing web UI's for services like databases, and this is considered a security best practice since the associated services won't be left exposed on public ports. Does docker swarm have something equivalent?

**Comment by BretFisher** [comment]
Not that I'm aware of.

**Comment by BretFisher** [comment]
To be fair, there is no web dashboard out of the box with Swarm, so there's nothing for the CLI to protect. The remote CLI should be TLS protected by the engine config. It's up to any 3rd party management tools like Docker EE, Portainer or Swarmpit to protect themselves (which they all do, with built-in web auth).
